lib/mix/lib/mix/project.ex

```diff
@@ -137,7 +137,7 @@ defmodule Mix.Project do
     [
       build_embedded: config[:build_embedded],
       build_per_environment: config[:build_per_environment],
-      consolidate_protocols: false,
+      consolidate_protocols: config[:consolidate_protocols],
```

Comment:
This should be false because we never want to consolidate it on the children!

Reply:
by keeping this as false, this PR will make it so that it never uses consolidated protocols with code coverage. So if this needs to remain false, then we should close this PR and turn it into an issue, since it's dependent on reading the `consolidate_protocols` value from the config from inside the context of the child application.

Reply:
I see. The issue is that by setting this to true, we will change other parts of Mix and cause it to consolidate when it wouldn't have before. I think the best option here is to check that the consolidation path is indeed in the load path. I will push a fix, thanks!